scripts/features.R

```diff
@@ -0,0 +1,125 @@
+
+# list all the fiss form names in the file
+form_names_l <- list.files(path = paste0('../../gis/mergin/',
+                                         dir_project),
+                           pattern = glob2rx('*site_2022*.gpkg'),
+                           full.names = T
+) %>%
+  stringr::str_subset('form_fiss_site_202209070609', negate = T) %>%
+  stringr::str_subset('form_fiss_site_202209100809', negate = T)
+
+
+
+# read all the forms into a list of dataframes using colwise to guess the column types
+# if we don't try to guess the col types we have issues later with the bind_rows join
+form <- form_names_l %>%
+  purrr::map(sf::st_read) %>%
+  purrr::map(plyr::colwise(type.convert)) %>%
+  # name the data.frames so we can add it later as a "source" column - we use basename to leave the filepath behind
+  purrr::set_names(nm = basename(form_names_l)) %>%
+  bind_rows(.id = 'source')
+
+
+# see the names of our form
+names(form)
+
+# select the columns we want
+
+features <- form %>%
+  select(local_name, feature_type, feature_height_m, feature_length_m, feature_height_length_method, feature_time) %>%
+  filter(!is.na(feature_type))
+
+features2 <- form %>%
+  select(local_name, feature_type_2, feature_height_2_m, feature_length_2_m, feature_height_length_method, feature_time_2) %>%
+  filter(!is.na(feature_type_2)) %>%
+  rename(feature_type = feature_type_2,
+         feature_height_m = feature_height_2_m,
+         feature_length_m = feature_length_2_m,
+         feature_time = feature_time_2) %>%
+  # there was bug in the mergin form where debris jam couldn't be added as feature type so it was included in length column
+  mutate(feature_type = case_when(feature_length_m == '2m debris jam' ~ 'LWD Jam', T ~ feature_type)) %>%
+  mutate(feature_length_m = case_when(feature_length_m == '2m debris jam' ~ '2', T ~ feature_length_m)) %>%
+  mutate(feature_length_m = as.numeric(feature_length_m))
+
+features3 <- form %>%
+  select(local_name, feature_type_3, feature_height_3_m, feature_length_3_m, feature_height_length_method, feature_time_3) %>%
+  filter(!is.na(feature_type_3)) %>%
+  rename(feature_type = feature_type_3,
+         feature_height_m = feature_height_3_m,
+         feature_length_m = feature_length_3_m,
+         feature_time = feature_time_3) %>%
+  # there was bug in the mergin form where debris jam couldn't be added as feature type so it was included in height column
+  mutate(feature_type = case_when(feature_height_m == '2 log jam' ~ 'LWD Jam', T ~ feature_type)) %>%
+  mutate(feature_height_m = case_when(feature_height_m == '2 log jam' ~ '2', T ~ feature_height_m)) %>%
+  mutate(feature_height_m = as.numeric(feature_height_m))
+
+features_final <- bind_rows(features, features2, features3) %>%
+  mutate(feature_time = lubridate::as_datetime(feature_time, tz="America/Vancouver"))
+
+# get UTMS of features that have times by joining to tracks
```

Comment:
this needs to be a bit more complicated because we have more than one person tracking at the same time.  This stuff might be best left for me after these inital stabs. I wen to war on this stuff over the last bunch of years so can get you up to speed much faster than you trying to battle through it yourself